File: src/data/Asset.ts
export type AssetStatus   = 'active' | 'inactive' | 'maintenance' | 'retired'
export type AssetCategory = 'ground_support' | 'passenger_services' | 'baggage_handling' | 'fueling' | 'aircraft_maintenance' | 'security' | 'facilities' | 'vehicles'

export interface Asset {
  id: number
  name: string
  serial_number: string
  category: AssetCategory
  status: AssetStatus
  location_id: number
  location_name?: string
  purchase_date: string
  purchase_price: number
  notes: string
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface CreateAssetPayload {
  name: string
  serial_number: string
  category: AssetCategory
  status: AssetStatus
  location_id: number
  purchase_date: string
  purchase_price: number
  notes?: string
}

export const ASSET_CATEGORIES: AssetCategory[] = [
  'ground_support', 'passenger_services', 'baggage_handling',
  'fueling', 'aircraft_maintenance', 'security', 'facilities', 'vehicles',
]

export const ASSET_STATUSES: AssetStatus[] = ['active', 'inactive', 'maintenance', 'retired']

export const CATEGORY_LABEL: Record<AssetCategory, string> = {
  ground_support:       'Ground Support',
  passenger_services:   'Passenger Services',
  baggage_handling:     'Baggage Handling',
  fueling:              'Fueling',
  aircraft_maintenance: 'Aircraft Maint.',
  security:             'Security',
  facilities:           'Facilities',
  vehicles:             'Vehicles',
}

export const STATUS_COLORS: Record<AssetStatus, string> = {
  active:      '#22c48a',
  inactive:    '#bbb',
  maintenance: '#f5a623',
  retired:     '#f05252',
}

File: backend/schemas/asset.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class AssetCreate(BaseModel):
    name:           str
    serial_number:  str
    category:       str
    status:         str = 'active'
    location_id:    int
    purchase_date:  str
    purchase_price: float = 0
    notes:          Optional[str] = ''


class AssetUpdate(BaseModel):
    name:           Optional[str]   = None
    category:       Optional[str]   = None
    status:         Optional[str]   = None
    location_id:    Optional[int]   = None
    purchase_price: Optional[float] = None
    notes:          Optional[str]   = None
    is_active:      Optional[bool]  = None


class AssetOut(AssetCreate):
    id:            int
    is_active:     bool
    location_name: Optional[str] = None
    created_at:    datetime
    updated_at:    datetime
    model_config = {"from_attributes": True}